app/controllers/users_controller.rb

```diff
@@ -24,6 +24,16 @@ def show
     end
   end
 
+  def edit
+  end
+
+  def update
+    current_user.attributes = user_params
+    current_user.save
+    flash[:succes] = "Your information has been updated."
+    redirect_to '/profile'
+  end
```

Comment:
I tried to think ahead and implement a check for when the user tries to update their info but leaves some fields blank but could not make it work. This is what I was trying in case anyone wants to give it a shot:

```
  def update
    current_user.attributes = user_params
    if current_user.save
      flash[:succes] = "Your information has been updated."
      redirect_to '/profile'
    else
      flash[:error] = current_user.errors.full_messages.uniq.to_sentence
      render :edit
    end
  end
```

Reply:
Thinking aloud: If the user is updating their profile, do they see a completed form that they can then delete/change the previous answers? Or is the whole thing blank and they have to "update" each field (even if they want a field or two to stay the same)...?

Reply:
They see a completed form initially! I'm just thinking in case they delete something and then forget to fill it out and click submit. 

**BUT** now that you've asked that question... I chose attributes method in `current_user.attributes = user_params` because it only updates the attributes that are passed in a leaves the other ones as is. Which means if someone for some reason submits the update form with blank fields it won't clear the record, it will just ignore it!

Reply:
I love this idea and it's a phenomenal use-case for a patch route.